cephfs: add support for VolumeCondition in NodeGetVolumeStats operation

nixpanic opened the pull request:
This is my work-in-progress state for supporting `VOLUME_CONDITION`, starting with CephFS.

Updates: #1356

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

nixpanic committed:
util: add health-checker for periodic filesystem checks

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
cephfs: enable VolumeCondition with new health-checker

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Madhu-1 commented:
@nixpanic This PR is part of work for https://github.com/ceph/ceph-csi/issues/1356?

nixpanic commented:
> @nixpanic This PR is part of work for #1356?

It surely could be!

nixpanic committed:
cephfs: use a new `data` subdirectory in NodePublishVolume

The new "data" subdirectory in the volume is bind-mounted inside the
container. The root of the volume is not available for applications, so
that Ceph-CSI can create a health-check file there and it is hidden from
the users.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
doc: Add initial design notes for the Health Checker

Signed-off-by: Niels de Vos <ndevos@ibm.com>
